portal/server/helpers/constructPayload.js

```diff
@@ -5,16 +5,24 @@
  * @param {Object} body Request body (req.body)
  * @param {Array} properties Interested properties to be added into payload
  * @param {Boolean} csrf Include CSRF token, defaulted to `true`
+ * @param {Array} unsetIfPropertyIsEmpty The condition to determine whether to delete the property
  * @returns {Object} Payload
  */
-const constructPayload = (body, properties, csrf = true) => {
+const constructPayload = (body, properties, unsetIfPropertyIsEmpty = [], csrf = true) => {
   let payload = {};
-
+  const newObj = { ...body };
```

Comment:
Using very generic variable names is not good practice, because variable purpose is not clear.

Maybe `bodyCopy` is better? feel free to think for better name with clear purpose.

Reply:
Changed to `bodyCopy`